scipy/optimize/_minimize.py

```diff
@@ -510,10 +513,13 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
         warn('Method %s does not use Hessian-vector product '
              'information (hessp).' % method, RuntimeWarning)
     # - constraints or bounds
-    if (meth in ('nelder-mead', 'powell', 'cg', 'bfgs', 'newton-cg', 'dogleg',
+    if (meth in ('nelder-mead', 'cg', 'bfgs', 'newton-cg', 'dogleg',
                  'trust-ncg') and (bounds is not None or np.any(constraints))):
         warn('Method %s cannot handle constraints nor bounds.' % method,
              RuntimeWarning)
+    if meth == "powell" and np.any(constraints):
```

Comment:
Why not just add `'powell'` to this set `('l-bfgs-b', 'tnc')` in the line below?

Reply:
Indeed, sorry I missed that! Now fixed.